tests/test_fixtures.py

```diff
@@ -32,19 +32,44 @@ def test_workspace_instance_scope_part2(workspace):
     assert workspace is workspace_instance[0]
 
 
-def test_hardware_node_add_remove(hardware):
+def _hardware_add_node(h, name, role) -> NodeBase:
     """
-    test the hardware fixture. Especially the handing of nodes
+    helper method to create a new hardware node
     """
-    nodes_length = len(hardware.nodes.keys())
+    nodes_length = len(h.nodes.keys())
     # create a new node
-    new_node = hardware.node_create('test1', NodeRole.WORKER, [])
+    new_node = h.node_create(name, role, [])
     # add the node to the hardware
-    hardware.node_add(new_node)
-    hardware.prepare_nodes(limit_to_nodes=[new_node])
+    h.node_add(new_node)
+    h.prepare_nodes(limit_to_nodes=[new_node])
     # we should have one more node now
-    assert len(hardware.nodes.keys()) == nodes_length+1
+    assert len(h.nodes.keys()) == nodes_length+1
+    return new_node
+
+
+def test_hardware_node_add_remove(hardware):
+    """
+    test the hardware fixture. Especially the handing of nodes
+    """
+    nodes_length = len(hardware.nodes.keys())
+    new_node = _hardware_add_node(hardware, 'test1', NodeRole.WORKER)
     # drop the node again
     hardware.node_remove(new_node)
     # we should have the old amount of nodes now
     assert len(hardware.nodes.keys()) == nodes_length
+
+
+def test_kubernetes_node_join(kubernetes):
+    """
+    test the kubernetes fixture
+    """
+    # current number of nodes in the k8s cluster
+    kubernetes_nodes_length = len(kubernetes.v1.list_node().items)
+    # create a new node
+    new_node = _hardware_add_node(kubernetes.hardware, 'test1',
+                                  NodeRole.WORKER)
+    # add the node to the kubernetes cluster
+    kubernetes.join([new_node])
+    import pdb; pdb.set_trace()
+    assert (len(kubernetes.v1.list_node().items) ==
+            kubernetes_nodes_length + 1)
```

Comment:
(nit) If you're re-rolling, could you please add a TODO to make a test the confirms that rook handles the added node correctly. For example, do we expect it to have an OSD created etc.

Reply:
this is k8s only - no rook involved.

Reply:
If a node joins the kubernetes cluster what does the rook operator do? If it's nothing, we should perhaps confirm that by making sure the number of OSD's etc is still correct.

Reply:
In this test, rook is not used at all. This is just testing the interface of our kubernetes fixture.

Reply:
I realise that, I just mean we should have another test for what happens to rook when adding a node in the future.

It's not important though, there are lots of things we should be testing.